utils/kubernetes/service.go

```diff
@@ -2,109 +2,74 @@ package kubernetes
 
 import (
 	"context"
+	"strings"
 
 	"github.com/layer5io/meshkit/utils"
-	coreV1 "k8s.io/api/core/v1"
-	v1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/kubernetes"
 )
 
-// GetServiceEndpoint returns the endpoint for the given service
-func (client *Client) GetServiceEndpoint(ctx context.Context, svcName, namespace string) (*utils.Endpoint, error) {
-	svc, err := client.KubeClient.CoreV1().Services(namespace).Get(ctx, svcName, v1.GetOptions{})
-	if err != nil {
-		return nil, ErrServiceDiscovery(err)
-	}
-
-	// Try loadbalancer endpoint
-	if endpoint := extractLoadBalancerEndpoint(svc); endpoint != nil {
-		return endpoint, nil
-	}
+type ServiceOptions struct {
+	Name         string
+	Namespace    string
+	PortSelector string
+	PortNumber   *int32
+	APIServerURL string
+}
 
-	// Try nodeport endpoint
-	nodes, err := client.KubeClient.CoreV1().Nodes().List(ctx, v1.ListOptions{})
+// GetServiceEndpoint returns the endpoint for the given service
+func GetServiceEndpoint(ctx context.Context, client *kubernetes.Clientset, opts *ServiceOptions) (*utils.Endpoint, error) {
```

Comment:
would be nice with some more explanation.

Reply:
@mgfeller  The function definition has been changed, and the output struct is also different. thus breaking change.

Reply:
@kumarabd ah ok thanks, but the functionality remains the same?

Reply:
yes @mgfeller